src/DotNetLightning.Core/Utils/HTLCOfferedExtensions.fs

```diff
@@ -0,0 +1,147 @@
+namespace DotNetLightning.Utils
+
+open System
+open NBitcoin
+open NBitcoin.BuilderExtensions
+open DotNetLightning.Utils
+
+open ResultUtils
+open ResultUtils.Portability
+open Newtonsoft.Json
+
+type HTLCOfferedParameters = {
+    LocalHtlcPubKey: HtlcPubKey
+    RemoteHtlcPubKey: HtlcPubKey
+}
+    with
+    static member TryExtractParameters (scriptPubKey: Script): Option<HTLCOfferedParameters> =
+        let ops =
+            scriptPubKey.ToOps()
+            // we have to collect it into a list and convert back to a seq
+            // because the IEnumerable that NBitcoin gives us is internally
+            // mutable.
+            |> List.ofSeq
+            |> Seq.ofList
+
+        Console.WriteLine(JsonConvert.SerializeObject(ops))
```

Comment:
remove Console.WriteLine? if you leave it there, explain why, and use 2 parenthesis instead of 4